driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -1387,7 +1387,7 @@ private static String generateClusterName() {
     final String clusterName;
     private boolean isInit;
     private volatile boolean isFullyInit;
-
+    private Exception initException;
```

Comment:
When I read this I thought that a `volatile` keyword would be required for this field, but it turns out that it is already guarded by `this` thanks to the `synchronized` block. 
_However_ : out of curiosity I checked usages of `isInit`, and IMO it should be volatile since it is accessed without holding the object's lock in `com.datastax.driver.core.Cluster.Manager.close()`. Wdyt?